browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -6765,9 +6765,23 @@ L.CanvasTileLayer = L.Layer.extend({
 			offset += len;
 		}
 
-	        if (imgData)
+		if (imgData)
 			ctx.putImageData(imgData, 0, 0);
 
+		// Partial fix for issue #5876 discard canvas contexts immediately
+		// WKWebView has a hardcoded memory limit for all canvas contexts
+		// so immediately convert the canvas to an image and release the
+		// canvas context's backing store.
+		// This bug only appears in the iOS app because most .png tiles are
+		// passed as raw data, not as "data:" strings, in the iOS app.
+		if (window.ThisIsTheiOSApp) {
+			tile.el = document.createElement('img');
+			tile.el.src = canvas.toDataURL();
```

Comment:
Ok - so 'toDataURL' does a compression as PNG - which uses deflate - which is hyper-slow and (I guess) single threaded. I would expect this to introduce a major negative performance impact =) not least because to get it back and apply a delta eg. typing 't' we need to de-compress, render to canvas, apply the 't' delta as pixels, and then compress back to png =)

Probably we should keep a cache of the last <N> tiles which we could manange in this method - and keep at least the last screen-full of them around in canvas form, and compress the rest to 'img' =)

Possibly we could also listen to some OS 'trim' event and compress them if the OS is desparate for space a s bonus =)

Thanks !